packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -0,0 +1,404 @@
+import * as React from 'react';
+import { IconButton } from '../../Button';
+import { Label } from '../../Label';
+import {
+  BaseComponent,
+  css,
+  getId,
+  KeyCodes
+} from '../../Utilities';
+import {
+  ISpinButton,
+  ISpinButtonProps
+} from './SpinButton.Props';
+import './SpinButton.scss';
+import { Position } from '../../utilities/positioning';
+
+export interface ISpinButtonState {
+  /**
+   * the value of the spin button
+   */
+  value?: string;
+
+  /**
+   * Are we spinning? Used in case we are spinning
+   * and the text field gets focus (we should stop spinning)
+   */
+  spinning?: boolean;
+
+  /**
+   * keyboard spin direction, used to style the up or down button
+   * as active when up/down arrow is pressed
+   *
+   * -1 == spinning down
+   * 0 == not spinning
+   * 1 == spinning up
+   */
+  keyboardSpinDirection?: number;
+}
+
+export class SpinButton extends BaseComponent<ISpinButtonProps, ISpinButtonState> implements ISpinButton {
+
+  public static defaultProps: ISpinButtonProps = {
+    step: 1,
+    min: 0,
+    max: 100,
+    disabled: false,
+    labelPosition: Position.start,
+    labelGapSpace: 10,
+    label: null,
+    incrementButtonIcon: 'ChevronUpSmall',
+    decrementButtonIcon: 'ChevronDownSmall'
+  };
+
+  private _input: HTMLInputElement;
+  private _inputId: string;
+  private _labelId: string;
+  private _lastValidValue: string;
+
+  private _onValidate?: (value: string) => string;
+  private _onIncrement?: (value: string) => string;
+  private _onDecrement?: (value: string) => string;
+
+  private _currentStepFunctionHandle: number;
+  private _stepDelay = 100;
+  private _formattedValidUnitOptions: string[] = [];
+
+  constructor(props?: ISpinButtonProps) {
+    super(props);
+
+    let value = props.value || props.defaultValue || String(props.min);
+    this._lastValidValue = value;
+
+    this.state = {
+      value: value,
+      spinning: false,
```

Comment:
If a field is not used in the render() function it should not be part of the state